Lib/sam.c

```diff
@@ -46,23 +58,322 @@ void ResetHandler();
+void TC0_Handler    ( void ) __attribute__ ((weak, alias("unused_isr")));
+void TC1_Handler    ( void ) __attribute__ ((weak, alias("unused_isr")));
+void TC2_Handler    ( void ) __attribute__ ((weak, alias("unused_isr")));
+#ifdef _SAM4S_TC1_INSTANCE_
+void TC3_Handler    ( void ) __attribute__ ((weak, alias("unused_isr")));
+#endif /* _SAM4S_TC1_INSTANCE_ */
+#ifdef _SAM4S_TC1_INSTANCE_
+void TC4_Handler    ( void ) __attribute__ ((weak, alias("unused_isr")));
+#endif /* _SAM4S_TC1_INSTANCE_ */
+#ifdef _SAM4S_TC1_INSTANCE_
+void TC5_Handler    ( void ) __attribute__ ((weak, alias("unused_isr")));
+#endif /* _SAM4S_TC1_INSTANCE_ */
+void ADC_Handler    ( void ) __attribute__ ((weak, alias("unused_isr")));
+#ifdef _SAM4S_DACC_INSTANCE_
+void DACC_Handler   ( void ) __attribute__ ((weak, alias("unused_isr")));
+#endif /* _SAM4S_DACC_INSTANCE_ */
+void PWM_Handler    ( void ) __attribute__ ((weak, alias("unused_isr")));
+void CRCCU_Handler  ( void ) __attribute__ ((weak, alias("unused_isr")));
+void ACC_Handler    ( void ) __attribute__ ((weak, alias("unused_isr")));
+void UDP_Handler    ( void ) __attribute__ ((weak, alias("unused_isr")));
+
+/* Exception Table */
+__attribute__ ((section(".vectors")))
+const DeviceVectors exception_table = {
+
+        /* Configure Initial Stack Pointer, using linker-generated symbols */
+        .pvStack = (void*) (&_estack),
+
+        .pfnReset_Handler      = (void*) ResetHandler,
+        .pfnNMI_Handler        = (void*) NMI_Handler,
+        .pfnHardFault_Handler  = (void*) HardFault_Handler,
+        .pfnMemManage_Handler  = (void*) MemManage_Handler,
+        .pfnBusFault_Handler   = (void*) BusFault_Handler,
+        .pfnUsageFault_Handler = (void*) UsageFault_Handler,
+        .pfnReserved1_Handler  = (void*) (0UL),          /* Reserved */
+        .pfnReserved2_Handler  = (void*) (0UL),          /* Reserved */
+        .pfnReserved3_Handler  = (void*) (0UL),          /* Reserved */
+        .pfnReserved4_Handler  = (void*) (0UL),          /* Reserved */
+        .pfnSVC_Handler        = (void*) SVC_Handler,
+        .pfnDebugMon_Handler   = (void*) DebugMon_Handler,
+        .pfnReserved5_Handler  = (void*) (0UL),          /* Reserved */
+        .pfnPendSV_Handler     = (void*) PendSV_Handler,
+        .pfnSysTick_Handler    = (void*) SysTick_Handler,
+
+        /* Configurable interrupts */
+        .pfnSUPC_Handler   = (void*) SUPC_Handler,   /* 0  Supply Controller */
+        .pfnRSTC_Handler   = (void*) RSTC_Handler,   /* 1  Reset Controller */
+        .pfnRTC_Handler    = (void*) RTC_Handler,    /* 2  Real Time Clock */
+        .pfnRTT_Handler    = (void*) RTT_Handler,    /* 3  Real Time Timer */
+        .pfnWDT_Handler    = (void*) WDT_Handler,    /* 4  Watchdog Timer */
+        .pfnPMC_Handler    = (void*) PMC_Handler,    /* 5  Power Management Controller */
+        .pfnEFC0_Handler   = (void*) EFC0_Handler,   /* 6  Enhanced Embedded Flash Controller 0 */
+#ifdef _SAM4S_EFC1_INSTANCE_
```

Comment:
Where do these defines come from? (in ASF specifically)

Reply:
CMSIS_Devices/ATMEL/sam4s/source/as_gcc/startup_sam4s.c

Reply:
They seem to come from including headers in an instance folder of the ASF. These files don't seem to have made it into the ExpiramentalCore-sam repo, but they are in other github projects such as this one - https://github.com/me-no-dev/Arduino-1/blob/master/hardware/arduino/sam/system/CMSIS/Device/ATMEL/sam3u/include/instance/instance_efc1.h